components/eamxx/src/physics/mam/eamxx_mam_aci_process_interface.cpp

```diff
@@ -0,0 +1,1299 @@
+  // Get physical constants
+  using C                      = physics::Constants<Real>;
+  static constexpr auto gravit = C::gravit;  // Gravity [m/s2]
+  // Gas constant for dry air [J/(kg*K) or J/Kg/K]
+  static constexpr auto rair = C::Rair;
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, 0u, top_lev), KOKKOS_LAMBDA(int kk) {
+        w0(icol, kk)  = 0;
+        rho(icol, kk) = -999.0;
+      });
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, top_lev, nlev), KOKKOS_LAMBDA(int kk) {
+        rho(icol, kk) = p_mid(icol, kk) / (rair * T_mid(icol, kk));
+        w0(icol, kk)  = -1.0 * omega(icol, kk) / (rho(icol, kk) * gravit);
+      });
+}
+void compute_w0_and_rho(haero::ThreadTeamPolicy team_policy,
+                        const mam_coupling::DryAtmosphere &dry_atmosphere,
+                        const int top_lev, const int nlev,
+                        // output
+                        MAMAci::view_2d w0, MAMAci::view_2d rho) {
+  MAMAci::const_view_2d omega = dry_atmosphere.omega;
+  MAMAci::const_view_2d T_mid = dry_atmosphere.T_mid;
+  MAMAci::const_view_2d p_mid = dry_atmosphere.p_mid;
+  Kokkos::parallel_for(
+      team_policy, KOKKOS_LAMBDA(const haero::ThreadTeam &team) {
+        const int icol = team.league_rank();
+        compute_w0_and_rho(team, omega, T_mid, p_mid, icol, top_lev, nlev,
+                           // output
+                           w0, rho);
+      });
+}
+
+void compute_values_at_interfaces(haero::ThreadTeamPolicy team_policy,
+                                  const MAMAci::const_view_2d var_mid,
+                                  const MAMAci::view_2d dz, const int nlev_,
+                                  // output
+                                  MAMAci::view_2d var_int) {
+  using CO = scream::ColumnOps<DefaultDevice, Real>;
+
+  Kokkos::parallel_for(
+      team_policy, KOKKOS_LAMBDA(const haero::ThreadTeam &team) {
+        const int icol = team.league_rank();
+
+        const auto var_mid_col = ekat::subview(var_mid, icol);
+        const auto var_int_col = ekat::subview(var_int, icol);
+        const auto dz_col      = ekat::subview(dz, icol);
+
+        const Real bc_top = var_mid_col(0);
+        const Real bc_bot = var_mid_col(nlev_ - 1);
+
+        CO::compute_interface_values_linear(team, nlev_, var_mid_col, dz_col,
+                                            bc_top, bc_bot, var_int_col);
+      });
+}
+
+KOKKOS_INLINE_FUNCTION
+void compute_tke_using_w_sec(const haero::ThreadTeam &team,
+                             const MAMAci::const_view_2d w_sec, const int icol,
+                             const int nlev,
+                             // output
+                             MAMAci::view_2d tke) {
+  Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, 0u, nlev + 1),
+      KOKKOS_LAMBDA(int kk) { tke(icol, kk) = (3.0 / 2.0) * w_sec(icol, kk); });
+}
+void compute_tke_using_w_sec(haero::ThreadTeamPolicy team_policy,
```

Comment:
This function name is the same as the above one (for KOKKOS). Is it a problem?

Reply:
No, C++ (like Fortran) allows function overloading. In Fortran, we have to use an "interface" but C++ allows it without interfaces.